albatross.ml

```diff
@@ -91,6 +95,94 @@ struct
             }
     | Error (`Msg err) -> Error err
 
+  let manifest_devices_match ~bridges ~block_devices binary =
+    let* mft : Solo5_elftool.mft = Solo5_elftool.query_manifest binary in
```

Comment:
Here we (unfortunately) need to convert the binary to a `Owee_buf.t` (bigarray):
```suggestion
  let manifest_devices_match ~bridges ~block_devices binary =
    let b = Bigarray.Array1.create Bigarray.Int8_unsigned Bigarray.c_layout (String.length b) in
    for i = 0 to String.length b - 1 do
      buf.{i} <- String.get_uint8 b i
    done;
    let binary = b in
    let* mft : Solo5_elftool.mft = Solo5_elftool.query_manifest binary in
```
Otherwise we get a type error. With this PR we should be able to do it without copying too much: https://git.robur.coop/robur/ocaml-solo5-elftool/pulls/2
So while this is very inefficient I'll submit a fix to that Soon™.